test/buffer/worker/client_test.go

```diff
@@ -0,0 +1,308 @@
+package worker
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/c2h5oh/datasize"
+	"github.com/keboola/go-client/pkg/keboola"
+	"github.com/keboola/go-client/pkg/request"
+	"github.com/spf13/cast"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/umisama/go-regexpcache"
+
+	bufferDesign "github.com/keboola/keboola-as-code/api/buffer"
+	apiModel "github.com/keboola/keboola-as-code/internal/pkg/service/buffer/api/gen/buffer"
+	apiServer "github.com/keboola/keboola-as-code/internal/pkg/service/buffer/api/gen/http/buffer/server"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/strhelper"
+)
+
+// apiClient is part of testSuite, it provides API requests to Buffer and Storage APIs.
+type apiClient struct {
+	t   *testing.T
+	ctx context.Context
+	ts  *testSuite
+}
+
+type Error struct {
+	Message string `json:"message"`
+}
+
+func (e Error) Error() string {
+	return e.Message
+}
+
+func newAPIClient(ts *testSuite) *apiClient {
+	return &apiClient{t: ts.t, ctx: ts.ctx, ts: ts}
+}
+
+func (c *apiClient) newRequest() request.HTTPRequest {
+	return request.NewHTTPRequest(c.ts.RandomAPINode().Client).WithError(&Error{})
+}
+
+func (c *apiClient) newRequestWithToken() request.HTTPRequest {
+	return c.newRequest().AndHeader("X-StorageAPI-Token", c.ts.project.StorageAPIToken().Token)
+}
+
+// TablePreview loads preview of the table, sorted by the "id" column.
+func (c *apiClient) TablePreview(tableID, sortBy string) *keboola.TablePreview {
+	c.t.Helper()
+	c.t.Logf(`loading preview of the table "%s" ...`, tableID)
+
```

Comment:
API client pre operacie nad Storage API (table preview) a Buffer API (vsetko ostatne).

`Goa` umoznuje go klienta vygenerovat, ale setupovat to pre tento E2E test mi prislo ako overkill (rovnako ako generovat to z OpenAPI). Takto je to aspon zamknute voci BC breakom, ... ak sa nieco rozbije v kode, tak testy spadnu.